parser.go

```diff
@@ -156,46 +173,49 @@ Returns a identifier with which the handler can be removed via UnregisterNetMess
 
 See also: RegisterEventHandler()
 */
-func (p *Parser) RegisterNetMessageHandler(handler interface{}) dp.HandlerIdentifier {
-	return p.msgDispatcher.RegisterHandler(handler)
+
+type HandlerIdentifier *int
+
+func (p *parser) RegisterNetMessageHandler(handler interface{}) HandlerIdentifier {
+	return HandlerIdentifier(p.msgDispatcher.RegisterHandler(handler))
 }
 
 // UnregisterNetMessageHandler removes a net-message handler via identifier.
 //
 // The identifier is returned at registration by RegisterNetMessageHandler().
-func (p *Parser) UnregisterNetMessageHandler(identifier dp.HandlerIdentifier) {
-	p.msgDispatcher.UnregisterHandler(identifier)
+func (p *parser) UnregisterNetMessageHandler(identifier HandlerIdentifier) {
+	p.msgDispatcher.UnregisterHandler(dp.HandlerIdentifier(identifier))
 }
 
-func (p *Parser) error() (err error) {
+func (p *parser) error() (err error) {
 	p.errLock.Lock()
 	err = p.err
 	p.errLock.Unlock()
 	return
 }
 
-func (p *Parser) setError(err error) {
+func (p *parser) setError(err error) {
 	if err != nil {
 		p.errLock.Lock()
 		p.err = err
 		p.errLock.Unlock()
 	}
 }
 
-// NewParser creates a new Parser with the default configuration.
+// NewParser creates a new parser with the default configuration.
 // The demostream io.Reader (e.g. os.File or bytes.Reader) must provide demo data in the '.DEM' format.
 //
 // See also: NewCustomParser() & DefaultParserConfig
-func NewParser(demostream io.Reader) *Parser {
+func NewParser(demostream io.Reader) *parser {
```

Comment:
I think this should return the interface

```suggestion
func NewParser(demostream io.Reader) Parser {
```